self-tests.yml

```diff
@@ -750,6 +789,51 @@ buildvariants:
       - name: ".agent .test"
       - name: ".cli .test"
 
+  - name: build-and-push
```

Comment:
do we want this main task to also be in its own variant, or should this one be in the main variant? (I thought long and hard about how to avoid the multiple variants but i see what you're saying haha, i also think that the "build-staging" tag is pretty elegant)

Reply:
Thanks! 😄 
An alternative could be to go back to having a single task which builds serially, and we could have different variants of it for non-signing and staging, but parallelizing it does seem like a substantive improvement (today it's been like 3 minutes to build the whole enchilada).

I'm okay to put the main one in the ubuntu variant, though I kind of like having it separate (though not that there are three of them), since it separates testing the thing from building it. It'll be good to be able to see at a glance when the build has finished. On the other hand, a lot of variants might be too much clutter on the patch configuration page. I'm thinking to leave the main one in its own variant.